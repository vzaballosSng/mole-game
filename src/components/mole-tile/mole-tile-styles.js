import { css } from 'lit';

export const moleStyles = css`
	.tile {
		width: 6.25rem;
		height: 6.25rem;
		border: 1px solid black;
    background-color: #000000c9;
	}

	.bg-mole {
    width: 6.25rem;
		height: 6.25rem;
		background-image: url('public/molebg.png');
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
	}

  .player-name {
    color: white;
    line-height: 2rem;
  }

  .tile p {
    color: black;
  }
`;
