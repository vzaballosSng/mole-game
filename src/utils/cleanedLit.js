// eslint-disable-next-line no-undef
const { prettyDOM } = require('@testing-library/dom');

export const cleanedLit = (htmlElement) => {
	const domHTML = prettyDOM(htmlElement, Infinity);
	const cleaned = domHTML.replace(/"<!--\?lit\$.*?-->"/g, '""');
	return cleaned;
};
export const cleanedLitElements = (htmlElements) => {
	const cleaned = htmlElements.map((htmlElement) => {
		return cleanedLit(htmlElement);
	});
	return cleaned;
};
