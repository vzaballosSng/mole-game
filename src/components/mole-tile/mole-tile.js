import { LitElement, html } from 'lit';
import { moleStyles } from './mole-tile-styles.js';

export class MoleTile extends LitElement {
  static get properties() {
    return {
      showMole: { type: Boolean, attribute: 'show-mole' },
      position: { type: Number },
    };
  }

  static get styles() {
    return [moleStyles];
  }

  constructor() {
    super();
    this.showMole = false;
  }

  render() {
    return html`
    <div class="tile">
      ${this.showMole ?  html`<div class="bg-mole"></div>` : ''}
    </div>`;
  }
}

window.customElements.define('mole-tile', MoleTile);
