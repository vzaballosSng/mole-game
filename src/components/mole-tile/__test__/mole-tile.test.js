import { html, fixture } from '@open-wc/testing';
import { expect } from '@jest/globals';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanedLit } from '../../../utils/cleanedLit.js';
import '../mole-tile.js';

expect.extend(toHaveNoViolations);

describe('mole-tile', () => {
	it('mole-tile matches the snapshot', async () => {
		const el = await fixture(
			html`<mole-tile></mole-tile>`
		);
		const dataCleaned = cleanedLit(el);
		expect(dataCleaned).toMatchSnapshot();
	});

	it('accessibility test', async () => {
		const el = await fixture(html`<mole-tile icon="person"></mole-tile>`);
		const results = await axe(el);
		expect(results).toHaveNoViolations();
	});
});
