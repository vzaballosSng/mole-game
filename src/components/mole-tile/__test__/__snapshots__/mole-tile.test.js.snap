// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mole-tile mole-tile matches the snapshot 1`] = `"[36m<mole-tile />[39m"`;
